builtin/logical/pki/path_config_acme.go

```diff
@@ -0,0 +1,151 @@
+			},
+			"allowed_issuers": {
+				Type:        framework.TypeString,
+				Description: `which issuers are allowed for use with ACME; by default, this will only be the primary (default) issuer`,
+				Default:     "default",
+			},
+			"allowed_roles": {
+				Type:        framework.TypeString,
+				Description: `which roles are allowed for use with ACME; by default these will be all roles matching our selection criteria`,
+				Default:     "*",
+			},
+			"default_role": {
+				Type:        framework.TypeString,
+				Description: `if not empty, the role to be used for non-role-qualified ACME requests; by default this will be empty, meaning ACME issuance will be equivalent to sign-verbatim,`,
+				Default:     "",
+			},
+			"allow_no_allowed_domains": {
+				Type:        framework.TypeBool,
+				Description: `whether ACME will allow the use of roles without any explicit list of allowed domains`,
+				Default:     false,
+			},
+			"allow_any_domain": {
+				Type:        framework.TypeBool,
+				Description: `whether ACME will allow the use of roles with allow_any_name=true set.`,
+			},
+		},
+
+		Operations: map[logical.Operation]framework.OperationHandler{
+			logical.ReadOperation: &framework.PathOperation{
+				DisplayAttrs: &framework.DisplayAttributes{
+					OperationSuffix: "acme-configuration",
+				},
+				Callback: b.pathAcmeRead,
+			},
+			logical.UpdateOperation: &framework.PathOperation{
+				Callback: b.pathAcmeWrite,
+				DisplayAttrs: &framework.DisplayAttributes{
+					OperationVerb:   "configure",
+					OperationSuffix: "acme",
+				},
+				// Read more about why these flags are set in backend.go.
+				ForwardPerformanceStandby:   true,
+				ForwardPerformanceSecondary: true,
+			},
+		},
+
+		HelpSynopsis:    pathConfigAcmeHelpSyn,
+		HelpDescription: pathConfigAcmeHelpDesc,
+	}
+}
+
+func (b *backend) pathAcmeRead(ctx context.Context, req *logical.Request, _ *framework.FieldData) (*logical.Response, error) {
+	sc := b.makeStorageContext(ctx, req.Storage)
+	config, err := sc.getAcmeConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	return genResponseFromAcmeConfig(config), nil
+}
+
+func genResponseFromAcmeConfig(config *acmeConfigEntry) *logical.Response {
+	response := &logical.Response{
+		Data: map[string]interface{}{
+			"allow_any_domain":         config.AllowAnyDomain,
+			"allowed_roles":            config.AllowedRoles,
+			"allow_no_allowed_domains": config.AllowNoAllowedDomains,
+			"allowed_issuers":          config.AllowedIssuers,
+			"default_role":             config.DefaultRole,
+			"enabled":                  config.Enabled,
+		},
+	}
+
+	// TODO: Add some nice warning if we are on a replication cluster and path isn't set
+
+	return response
+}
+
+func (b *backend) pathAcmeWrite(ctx context.Context, req *logical.Request, _ *framework.FieldData) (*logical.Response, error) {
+	sc := b.makeStorageContext(ctx, req.Storage)
+	config, err := sc.getAcmeConfig()
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
It looks like this doesn't do request parsing and updating of the config?

Reply:
oops